def/typescript.ts

```diff
@@ -15,7 +15,7 @@ export default function (fork: Fork) {
   var def = types.Type.def;
   var or = types.Type.or;
   var defaults = fork.use(sharedPlugin).defaults;
-  var StringLiteral = new types.Type(function (value, deep) {
```

Comment:
Makes sense. Now that we're exporting something that's not constructable, I wonder if it would simplify things to use plain objects and interfaces vs abstract classes and concrete subclasses.